newrelic/resource_newrelic_nrql_alert_condition.go

```diff
@@ -13,6 +13,102 @@ import (
+			"priority": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "critical",
+				Description:  "One of (critical, warning). Defaults to 'critical'. At least one condition term must have priority set to 'critical'.",
+				ValidateFunc: validation.StringInSlice([]string{"critical", "warning"}, false),
+			},
+			"threshold": {
+				Type:         schema.TypeFloat,
+				Required:     true,
+				Description:  "Must be 0 or greater. For baseline conditions must be in range [1, 1000].",
+				ValidateFunc: float64Gte(0.0),
+			},
+			// Does not exist in NerdGraph. Equivalent to `threshold_occurrences`,
+			// but with different wording.
+			"time_function": {
+				Deprecated:    "use `threshold_occurrences` attribute instead",
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "Valid values are: 'all' or 'any'",
+				ConflictsWith: []string{"term.0.threshold_occurrences"},
+				ValidateFunc:  validation.StringInSlice([]string{"all", "any"}, false),
+			},
+
+			// NerdGraph only. Equivalent to `time_function`,
+			// but with slightly different wording.
+			// i.e. `any` (old) vs `AT_LEAST_ONCE` (new)
+			"threshold_occurrences": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The criteria for how many data points must be in violation for the specified threshold duration. Valid values are: 'ALL' or 'AT_LEAST_ONCE' (case insensitive).",
+				ConflictsWith: []string{"term.0.time_function"},
+				ValidateFunc:  validation.StringInSlice([]string{"ALL", "AT_LEAST_ONCE"}, true),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return strings.EqualFold(old, new) // Case fold this attribute when diffing
+				},
+			},
+			// NerdGraph only. Equivalent to `duration`, but in seconds
+			"threshold_duration": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Description:   "The duration of time, in seconds, that the threshold must violate for in order to create a violation. Value must be a multiple of 60 and within 120-3600 seconds for baseline conditions and 120-7200 seconds for static conditions.",
+				ConflictsWith: []string{"term.0.duration"},
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					v := val.(int)
+
+					// Value must be a factor of 60.
+					if v%60 != 0 {
+						errs = append(errs, fmt.Errorf("%q must be a factor of 60, got: %d", key, v))
+					}
+
+					// This validation is a top-level validation check.
+					// Baseline conditions must be within range [120, 3600].
+					// Baseline condition validation lives in the "expand" functions.
+					if v < 120 || v > 7200 {
+						errs = append(errs, fmt.Errorf("%q must be between 120 and 7200 inclusive, got: %d", key, v))
+					}
+
+					return
+				},
+			},
+		},
+	}
+}
+
+func namedPrioritySchema() *schema.Resource {
+	rec := termSchema()
+	delete(rec.Schema, "priority")
```

Comment:
Thoughts on removing `priority` from the base term schema instead? And then adding the attribute in `termSchema`? Feels a little more future facing.

Reply:
Oh, as in add `priority` only where it's needed, rather than delete where it's not?

Reply:
Maybe functions like 

```go
termSchema() {
  // The new hotness (moved cheese)
}

termSchemaDeprecated() {
   // add priority here
}
```